graylog2-server/src/main/java/org/graylog/plugins/map/geoip/MaxmindDataAdapter.java

```diff
@@ -245,6 +254,55 @@ protected LookupResult doGet(Object key) {
                     LOG.warn("Unable to look up ASN data for IP address {}, returning empty result.", addr, e);
                     return LookupResult.empty();
                 }
+            case IPINFO_ASN:
+                try {
+                    final IPinfoASN ipInfo = reader.ipInfoASN(addr);
+                    if (ipInfo == null) {
+                        LOG.debug("No IPinfo location data for IP address {}, returning empty result.", addr);
+                        return LookupResult.empty();
+                    }
+
+                    return LookupResult.multi(ipInfo.asnNumeric(), ImmutableMap.builder()
+                            .put("asn", ipInfo.asn())
+                            .put("asn_numeric", ipInfo.asnNumeric())
+                            .put("name", ipInfo.name())
+                            .put("domain", ipInfo.domain())
+                            .put("type", ipInfo.type())
+                            .put("route", ipInfo.route())
+                            .build());
+
+                } catch (AddressNotFoundException e) {
+                    LOG.debug("Unable to look up IPinfo ASN data for IP address {}, returning empty result.", addr, e);
+                    return LookupResult.empty();
+                } catch (Exception e) {
+                    LOG.warn("Unable to look up IPinfo ASN data for IP address {}, returning empty result.", addr, e);
+                    return LookupResult.empty();
+                }
+            case IPINFO_STANDARD_LOCATION:
+                try {
+                    final IPinfoStandardLocation ipInfo = reader.ipInfoStandardLocation(addr);
+                    if (ipInfo == null) {
+                        LOG.debug("No IPinfo location data for IP address {}, returning empty result.", addr);
+                        return LookupResult.empty();
+                    }
+
+                    return LookupResult.multi(ipInfo.coordinates(), ImmutableMap.builder()
```

Comment:
Can we be sure that we get data for all these fields back from the DB? E.g. is `city` guaranteed to be always provided?  I couldn't find an example for incomplete data, but what do you think about being a bit more cautious? We could make fields in `IPinfoStandardLocation` nullable/optional and construct the map with only those values which are present here.

Reply:
Aye, I made all properties nullable. :+1: